pkg/cloudprovider/aws/aws.go

```diff
@@ -429,6 +436,17 @@ func (aws *AWSCloud) ExternalID(name string) (string, error) {
 	return *inst.InstanceID, nil
 }
 
+// InstanceID returns the cloud provider ID of the specified instance.
+func (aws *AWSCloud) InstanceID(name string) (string, error) {
+	inst, err := aws.getInstancesByDnsName(name)
+	if err != nil {
+		return "", err
+	}
+	// In the future it is possible to also return an endpoint as:
+	// <endpoint>/<zone>/<instanceid>
+	return "/" + *inst.Placement.AvailabilityZone + "/" + *inst.InstanceID, nil
```

Comment:
Should InstanceID return a name with a leading slash (here and all other cloudproviders)? You prefix it with ProviderName() followed by `://` to produce the providerID. Also, please document the expectation in the interface declaration.